actioncable/app/javascript/action_cable/connection_monitor.js

```diff
@@ -75,26 +73,33 @@ class ConnectionMonitor {
   }
 
   getPollInterval() {
-    const {min, max, multiplier} = this.constructor.pollInterval
-    const interval = multiplier * Math.log(this.reconnectAttempts + 1)
-    return Math.round(clamp(interval, min, max) * 1000)
+    const interval = this.constructor.staleThreshold
+    const backoff = this.constructor.reconnectionBackoff
+    const exponent = Math.min(this.reconnectAttempts, 10)
+    const jitter = this.reconnectAttempts > 0 ? backoff - 1 : 1
+    const seconds = interval * Math.pow(backoff, exponent) * (1 + jitter * Math.random())
+    return seconds * 1000
```

Comment:
I keep feeling like we can make this easier to read, I haven't quite got it, but I think this is clearer to see the components:

```suggestion
  getPollInterval() {
    const { staleThreshold, reconnectionBackoffPercent } = this.constructor

    const exponentialBackoff = Math.pow(1 + reconnectionBackoffPercent, Math.min(this.reconnectAttempts, 10))
    const jitter = this.reconnectAttempts > 0 ? jitter({ percent: reconnectionBackoffPercent }) : jitter()
    return staleThreshold * exponentialBackoff * jitter * 1000
  }
```

Then later in the file:

```javascript
const reconnectionBackoffPercent = 0.15
const jitter = ({ percent } = 1) => Math.random() * percent + 1 // I can't remember the JS keyword argument syntax 😂
```

I keep wanting to call `reconnectionBackoff` `reconnectionBackoffPercent` even though it's not. I can't remember what the term for number between 0 to 1 when it's not in percent?

Reply:
@kaspth I incorporated most of your suggestions.  The separate `jitter` function felt a little off to me, though.  Please let me know what you think.